### PR TITLE
Run CircleCI jobs in seperate workflows to enable rerunning individually

### PR DESCRIPTION
**Pull request by akihironitta**
## What does this PR do?
Enables individually rerunning the two circleci jobs, `build-Docs` and `TPU-tests` by dividing the workflow into two for each job.

~Just trying... not 100% sure if it works.~ It seems working correctly :)

### Does your PR introduce any breaking changes? If yes, please list them.
None

## Before submitting

- [RFC] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [n/a] Did you make sure to **update the documentation** with your changes? (if necessary)
- [n/a] Did you write any **new necessary tests**? (not for typos and docs)
- [n/a] Did you verify new and **existing tests pass** locally with your changes?
- [n/a] Did you list all the **breaking changes** introduced by this pull request?
- [n/a] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

For reference: https://circleci.com/docs/2.0/configuration-reference/#workflows

cc @borda @carmocca @akihironitta

### CODE REVIEW

**Review by Borda — APPROVED** [review]

**Comment by akihironitta** [comment]
@Borda I think I have to ask someone with a higher privilege to change the setting in this repo. Could you update it after (or before?) this PR lands if you can?

Screen Shot 2022-03-25 at 17 30 22

**Comment by Borda** [comment]
> I think I have to ask someone with a higher privilege to change the setting in this repo. Could you update it after (or before?) this PR lands if you can?

yes, need to ping @lexierule

File: .circleci/config.yml [hunk 1]
```diff
@@ -157,7 +157,9 @@ jobs:
 
 workflows:
   version: 2
-  ci-tests:
+  build-docs:
     jobs:
       - build-Docs
+  test-on-tpus:
```
Comment (carmocca) [inline .circleci/config.yml:163]: Note that they can currently be run separately inside the CircleCI website 
Screenshot 2022-03-25 at 11 56 48

**Review by carmocca — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (akihironitta) [reply .circleci/config.yml:163]: Can they really? I tried that before but both `TPU-tests` and `build-Docs` got rerun in the end...

Comment (carmocca) [reply .circleci/config.yml:163]: I think so, but this PR is still useful so we can do it from GitHub

**Review by awaelchli — APPROVED** [review]

**Comment by awaelchli** [comment]
something is blocked :( 
image

**Comment by Borda** [comment]
> something is blocked :( image

yep, we know, need Admin to update the required checks and the rename drops the original and creates two new ones instead... :)

**Comment by akihironitta** [comment]
Hi @Borda @carmocca, this is nothing urgent, but could you ping @lexierule to update the setting for required jobs so that we can merge this PR?

**Comment by akihironitta** [comment]
@Borda Any idea why the old one is still required as seen in #12614?

Screen Shot 2022-04-05 at 13 18 51

**Comment by Borda** [comment]
> Any idea why the old one is still required as seen in #12614?

this was merged, but still, need to sync with @lexierule :rabbit:

**Pull request merged**